src/components/interaction-designer/block-editors/ChoicesBuilder.vue

```diff
@@ -0,0 +1,143 @@
+  </div>
+</template>
+
+<script lang="ts">
+import {get, isEmpty, last} from 'lodash'
+import {findOrGenerateStubbedVariantOn} from '@/store/flow/resource'
+import {Component, Prop} from 'vue-property-decorator'
+import {mixins} from 'vue-class-component'
+import Lang from '@/lib/filters/lang'
+import {IBlock, IBlockExit, IFlow, IResource, SupportedContentType, SupportedMode} from '@floip/flow-runner'
+import {namespace} from 'vuex-class'
+import ResourceVariantTextEditor from '@/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue'
+import {ISelectOneResponseBlock} from '@floip/flow-runner/src/model/block/ISelectOneResponseBlock'
+import SelectOneStore, {BLOCK_TYPE} from '@/store/flow/block-types/MobilePrimitives_SelectOneResponseBlockStore'
+import {IdGeneratorUuidV4} from '@floip/flow-runner/dist/domain/IdGeneratorUuidV4'
+import Vue from 'vue'
+
+const flowVuexNamespace = namespace('flow')
+const blockVuexNamespace = namespace(`flow/${BLOCK_TYPE}`)
+
+@Component({
+  components: {
+    ResourceVariantTextEditor,
+
+  },
+})
+class ChoicesBuilder extends mixins(Lang) {
+  @Prop() readonly block!: ISelectOneResponseBlock
+
+  draftResource: IResource | null = null
+
+  SupportedContentType = SupportedContentType
+  SupportedMode = SupportedMode
+  findOrGenerateStubbedVariantOn = findOrGenerateStubbedVariantOn
+
+  get choiceKeys(): string[] {
+    // todo: my best guess for ordering is resources ordering? :/
+    return Object.keys(this.block.config.choices)
+  }
+
+  created(): void {
+    this.generateDraftResource()
+  }
+
+  async generateDraftResource(): Promise<void> {
+    this.draftResource = await this.resource_createWith({ // todo: abstract as createResourceForVanillaChoice()
+      props: {uuid: await (new IdGeneratorUuidV4()).generate()},
+    })
+  }
+
+  async addDraftResourceToChoices(): Promise<void> {
+    if (this.draftResource == null) {
+      console.warn(
+        'interaction-designer/block-editors/ChoicesBuilder',
+        'Unable to add absent draft resource to block choices.',
+        {block: this.block})
+      return
+    }
+
+    this.resource_add({resource: this.draftResource})
+    this.addChoiceByResourceIdTo({blockId: this.block.uuid, resourceId: this.draftResource.uuid})
+
+    this.$nextTick(() =>
+      this.focusInputElFor(last(this.$refs.choices as Vue[])))
+
+    await this.generateDraftResource()
+  }
+
+  focusInputElFor(editor?: Vue): void {
+    (editor?.$refs.input as HTMLInputElement).focus()
+  }
+
+  handleExistingResourceVariantChangedFor(resourceId: IResource['uuid'], choiceIndex: number, resource: IResource): void {
+    const isLast = choiceIndex === this.choiceKeys.length - 1
+    const hasEmptyValue = isEmpty(get(resource.values[0], 'value'))
+
+    if (isLast && hasEmptyValue) {
+      // todo: clean up resource, but should we first check for references?
```

Comment:
same in this TODO
is it possible to track it in a ticket if it is still relevant ?